compositor_common/src/validators.rs

```diff
@@ -206,6 +217,40 @@ impl SceneSpec {
 
         Ok(())
     }
+
+    fn validate_builtin_transformations(
+        &self,
+        transform_nodes: &HashMap<&NodeId, &NodeSpec>,
+    ) -> Result<(), SpecValidationError> {
+        for (&node_id, &node_spec) in transform_nodes {
+            if let NodeParams::Builtin { transformation, .. } = &node_spec.params {
+                match transformation {
+                    BuiltinTransformationSpec::TransformToResolution(_) => {
+                        if node_spec.input_pads.len() == 1 {
+                            return Err(SpecValidationError::InvalidBuiltinParams(
+                                node_id.clone(),
+                                transformation.clone(),
+                                Arc::from("input_pads should contain only a single node"),
+                            ));
+                        }
+                    }
```

Comment:
This should be replaced with the more general system for specifying possible valid numbers of inputs